CodeEdit/Features/Documents/WorkspaceDocument/WorkspaceDocument.swift

```diff
@@ -43,11 +43,26 @@ final class WorkspaceDocument: NSDocument, ObservableObject, NSToolbarDelegate {
     var workspaceSettingsManager: CEWorkspaceSettings?
     var taskNotificationHandler: TaskNotificationHandler = TaskNotificationHandler()
 
+    @Published var notificationPanel = NotificationPanelViewModel()
+    private var notificationPanelSubscription: AnyCancellable?
+
     private var cancellables = Set<AnyCancellable>()
 
+    override init() {
+        super.init()
+
+        // Observe changes to notification panel
+        notificationPanelSubscription = notificationPanel.objectWillChange
```

Comment:
Would it be cleaner to use the `cancellables` variable for this, since cleaning it up is already handled in the `deinit`?

Reply:
Good catch!